src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -338,39 +379,47 @@ public void cleanup(Long id, boolean deleteRepositories) {
         for (StudentParticipation participation : exercise.getStudentParticipations()) {
             ProgrammingExerciseStudentParticipation studentParticipation = (ProgrammingExerciseStudentParticipation) participation;
             try {
-                if (studentParticipation.getRepositoryUrl() == null || studentParticipation.getStudent() == null
-                        || !studentIds.contains(studentParticipation.getStudent().getLogin())) {
+                if (!allStudents && (studentParticipation.getRepositoryUrl() == null || studentParticipation.getStudent() == null
```

Comment:
I don't like the design of this whole method, I'll pick the argument `allStudents` vs `studentIds` here as an example and explain what we could improve:
- Instead of having two arguments, we could just pass in the concerned participations as a list. Then the method knows which repositories to export and only has to iterate over them
- A method that calls this method can then only load the participations from the db that are needed for the export operation

The rest of the parameters (e.g. `filterLateSubmissions`) could be grouped into a new Class called `RepositoryExportOptions`. This way the method has less parameters and we can document the options better.